dbms/src/Interpreters/Context.cpp

```diff
@@ -1449,15 +1447,31 @@ BackgroundProcessingPool & Context::getBackgroundPool()
     return *shared->background_pool;
 }
 
+BackgroundProcessingPool & Context::initializeBackgroundPool(UInt16 pool_size)
+{
+    auto lock = getLock();
+    if (!shared->background_pool)
+        shared->background_pool = std::make_shared<BackgroundProcessingPool>(pool_size);
+    return *shared->background_pool;
+}
+
 BackgroundProcessingPool & Context::getBlockableBackgroundPool()
 {
-    // TODO: choose a better thread pool size and maybe a better name for the pool
+    // TODO: maybe a better name for the pool
     auto lock = getLock();
     if (!shared->blockable_background_pool)
         shared->blockable_background_pool = std::make_shared<BackgroundProcessingPool>(settings.background_pool_size);
     return *shared->blockable_background_pool;
 }
 
+BackgroundProcessingPool & Context::initializeBlockableBackgroundPool(UInt16 pool_size)
+{
+    auto lock = getLock();
+    if (!shared->blockable_background_pool)
+        shared->blockable_background_pool = std::make_shared<BackgroundProcessingPool>(pool_size);
```

Comment:
ditto

Reply:
done, apply the check inside `BackgroundProcessingPool::BackgroundProcessingPool(int size_)`